ImperatorToCK3/Source/Imperator/ImperatorWorld.cpp

```diff
@@ -15,6 +15,27 @@ namespace fs = std::filesystem;
 ImperatorWorld::World::World(const Configuration& theConfiguration)
 {
 	LOG(LogLevel::Info) << "*** Hello Imperator, Roma Invicta! ***";
+	
+	// parse the genes file
+	genes = GenesDB(theConfiguration.getImperatorPath() + "/game/common/genes/00_genes.txt");
```

Comment:
shove this entire block in genedb constructor, or a separate world function. You'll want to keep the parsing part as clean as possible, otherwise over time it'll grow spaghetti tentacles.

Reply:
✔️ Added `void parseGenes(const Configuration& theConfiguration)`